### PR TITLE
Implemented ObjectGuid support for AzerothCore.

### PR DESCRIPTION
**Pull request by UltraNix**
Please - merge it once https://github.com/azerothcore/azerothcore-wotlk/pull/4885 is merged.

### CODE REVIEW

File: PlayerMethods.h [hunk 1]
```diff
@@ -2278,11 +2278,10 @@ namespace LuaPlayer
...
 
         WorldPacket data(MSG_AUCTION_HELLO, 12);
+        data << unit->GET_GUID();
 #ifdef TRINITY
-        data << uint64(unit->GetGUID().GetCounter());
         data << uint32(ahEntry->ID);
 #else
-        data << uint64(unit->GetGUIDLow());
         data << uint32(ahEntry->houseId);
 #endif
```
Comment (Rochet2) [inline PlayerMethods.h:2286]: Same here

File: PlayerMethods.h [hunk 2]
```diff
@@ -2209,7 +2209,7 @@ namespace LuaPlayer
         player->SetSummonPoint(summoner->GetMapId(), x, y, z);
 
         WorldPacket data(SMSG_SUMMON_REQUEST, 8 + 4 + 4);
-        data << uint64(summoner->GetGUIDLow());
+        data << summoner->GET_GUID();
```
Comment (Rochet2) [inline PlayerMethods.h:2212]: Guid is a different number from guid low. One of the codes likely has a bug.

File: PlayerMethods.h [hunk 3]
```diff
@@ -2434,7 +2433,7 @@ namespace LuaPlayer
 #ifndef AZEROTHCORE
             player->UnbindInstance(map, (Difficulty)difficulty);
 #else
-            sInstanceSaveMgr->PlayerUnbindInstance(player->GetGUIDLow(), map, Difficulty(difficulty), true, player);
+            sInstanceSaveMgr->PlayerUnbindInstance(player->GetGUID(), map, Difficulty(difficulty), true, player);
```
Comment (Rochet2) [inline PlayerMethods.h:2436]: potentially same here

File: PlayerMethods.h [hunk 4]
```diff
@@ -2459,12 +2458,12 @@ namespace LuaPlayer
...
         for (uint8 i = 0; i < MAX_DIFFICULTY; ++i)
         {
-            const BoundInstancesMap& binds = sInstanceSaveMgr->PlayerGetBoundInstances(player->GetGUIDLow(), Difficulty(i));
+            const BoundInstancesMap& binds = sInstanceSaveMgr->PlayerGetBoundInstances(player->GetGUID(), Difficulty(i));
             for (BoundInstancesMap::const_iterator itr = binds.begin(); itr != binds.end();)
             {
                 if (itr->first != player->GetMapId())
                 {
-                    sInstanceSaveMgr->PlayerUnbindInstance(player->GetGUIDLow(), itr->first, Difficulty(i), true, player);
+                    sInstanceSaveMgr->PlayerUnbindInstance(player->GetGUID(), itr->first, Difficulty(i), true, player);
```
Comment (Rochet2) [inline PlayerMethods.h:2466]: and here

File: PlayerMethods.h [hunk 5]
```diff
@@ -3740,9 +3735,9 @@ namespace LuaPlayer
     /**
      * Adds a new item to the gossip menu shown to the [Player] on next call to [Player:GossipSendMenu].
      *
-     * sender and intid are numbers which are passed directly to the gossip selection handler. Internally they are partly used for the database gossip handling.  
-     * code specifies whether to show a box to insert text to. The player inserted text is passed to the gossip selection handler.  
-     * money specifies an amount of money the player needs to have to click the option. An error message is shown if the player doesn't have enough money.  
+     * sender and intid are numbers which are passed directly to the gossip selection handler. Internally they are partly used for the database gossip handling.
+     * code specifies whether to show a box to insert text to. The player inserted text is passed to the gossip selection handler.
+     * money specifies an amount of money the player needs to have to click the option. An error message is shown if the player doesn't have enough money.
```
Comment (Rochet2) [inline PlayerMethods.h:3740]: Whitespace should be left alone really. These changes actually change the documentation as far as I know as double space at end of line is considered a new line in markdown.
Either undo this or use <br /> etc.

File: ServerHooks.cpp [hunk 6]
```diff
@@ -140,19 +140,23 @@ void Eluna::OnChange(Weather* /*weather*/, uint32 zone, WeatherState state, floa
...
+#else
     Player* owner = eObjectAccessor()FindPlayer(MAKE_NEW_GUID(entry->owner, 0, HIGHGUID_PLAYER));
+#endif
+
 #ifdef TRINITY
     Item* item = eAuctionMgr->GetAItem(entry->itemGUIDLow);
     uint32 expiretime = entry->expire_time;
 #elif AZEROTHCORE
-    Item* item = eAuctionMgr->GetAItem(entry->item_guidlow);
+    Item* item = eAuctionMgr->GetAItem(entry->item_guid);
```
Comment (Rochet2) [inline ServerHooks.cpp:153]: guid thing here too

File: ServerHooks.cpp [hunk 7]
```diff
@@ -170,12 +174,17 @@ void Eluna::OnAdd(AuctionHouseObject* /*ah*/, AuctionEntry* entry)
...
```
Comment (Rochet2) [inline ServerHooks.cpp:187]: guid thing here too

File: ServerHooks.cpp [hunk 8]
```diff
@@ -200,12 +209,17 @@ void Eluna::OnRemove(AuctionHouseObject* /*ah*/, AuctionEntry* entry)
...
```
Comment (Rochet2) [inline ServerHooks.cpp:222]: guid thing here too

File: ServerHooks.cpp [hunk 9]
```diff
@@ -230,12 +244,17 @@ void Eluna::OnSuccessful(AuctionHouseObject* /*ah*/, AuctionEntry* entry)
...
```
Comment (Rochet2) [inline ServerHooks.cpp:257]: guid thing here too

**Comment by Rochet2** [comment]
I think overall it looks fine.
Slightly mixed on the addition of ObjectGuid to the lua interface as it is converted into uint64 anyways though, but I guess its fine.

File: PlayerMethods.h (hunk 3 above)
Comment (UltraNix) [reply PlayerMethods.h:2436]: This line is specific to AzerothCore only and it's okay.

File: PlayerMethods.h (hunk 2 above)
Comment (UltraNix) [reply PlayerMethods.h:2212]: I know. Low part of guid is wrong here - raw value is needed (you can observe that by looking at the same packet in different cores). But for players, low part = raw value (highguid for players is 0) - no bug here.

File: PlayerMethods.h (hunk 1 above)
Comment (UltraNix) [reply PlayerMethods.h:2286]: Explained below.

File: ServerHooks.cpp (hunk 6 above)
Comment (UltraNix) [reply ServerHooks.cpp:153]: This is how it is done in AzerothCore and that line is only for AzerothCore.

**Comment by FrancescoBorzi** [comment]
the PR has been merged on the AC side, so please restart the build here so we can merge this too

**Comment by UltraNix** [comment]
As @FrancescoBorzi said - ready to me merged.

**Pull request merged**